hipscat/catalog.py

```diff
@@ -254,116 +260,312 @@ def cross_match(self, othercat=None, c1_cols={}, c2_cols={}, n_neighbors=1, dthr
 
-        if self.output_dir is not None:
-            outputdir_files = os.listdir(self.output_dir)
-            maps = [x for x in outputdir_files if f'{self.catname}_order' in x and 'hpmap.fits' in x]
-            if len(maps):
-                map0 = maps[0]
-                mapFn = os.path.join(self.output_dir, map0)
-                highest_order = int(mapFn.split('order')[1].split('_')[0])
-                nside = hp.order2nside(highest_order)
-                vec = hp.ang2vec(ra, dec, lonlat=True)
-                rad = np.radians(radius)
-                pixels_to_query = hp.query_disc(nside, vec, rad, nest=True)
-
-                img = hp.read_map(mapFn)
-                maxN = max(np.log10(img+1))+1
-                img[pixels_to_query] = maxN
-                fig = plt.figure(figsize=figsize)
-                return hp.mollview(np.log10(img+1), fig=fig, title=f'Cone search of {self.catname}', nest=True)
-            else:
-                assert False, 'map not found. hips_import() needs to be (re-) ran'
-        else:
-            assert False, 'hipscat output_dir not found. hips_import() needs to be (re-) ran'
+        #Look for the output_dir created from 
+        if self.output_dir is None:
+            raise FileNotFoundError('hipscat output_dir not found. hips_import() needs to be (re-) ran')
+
+        #look for the pixel map fits file
+        outputdir_files = os.listdir(self.output_dir)
+        maps = [x for x in outputdir_files if f'{self.catname}_order' in x and 'hpmap.fits' in x]
+        if len(maps) == 0:
+            raise FileNotFoundError('map not found. hips_import() needs to be (re-) ran')
+
+        #grab the first one in the list
+        map0 = maps[0]
+        mapFn = os.path.join(self.output_dir, map0)
+
+        #grab the required parameters for hp.query(disc)
+        highest_order = int(mapFn.split('order')[1].split('_')[0])
+        nside = hp.order2nside(highest_order)
+        vec = hp.ang2vec(ra, dec, lonlat=True)
+        rad = np.radians(radius)
+        pixels_to_query = hp.query_disc(nside, vec, rad, nest=True)
+
+        img = hp.read_map(mapFn)
+        maxN = max(np.log10(img+1))+1
+
+        #plot pencil beam. Maybe change the maxN to a value that will 
+        #   never occur in np.log10(img+1).. maybe a negative number
+        img[pixels_to_query] = maxN
+        fig = plt.figure(figsize=figsize)
+        return hp.mollview(np.log10(img+1), fig=fig, title=f'Cone search of {self.catname}', nest=True)
+
+
+    def visualize_cross_match(self, othercat, figsize=(5,5)):
+        '''
+            Returns hp.mollview of the overlap when crossmatching two catalogs
+
+            inputs:
+                othercat=hipscat.Catalog()
+                figsuze=Tuple(x,y)
+
+        Visualize from notebook
+        '''
+        raise NotImplementedError('catalog.visualize_cross_match() not implemented yet')
+
+
+    def visualize_sources(self, figsize=(5,5)):
```

Comment:
There's a ton of copy/pasta after this line.

Reply:
not sure how that got there. removed